src/main/java/com/efficio/fieldbook/web/trial/controller/AdvancingController.java

```diff
@@ -728,7 +730,19 @@ public int countPlots(@PathVariable final String ids) throws MiddlewareQueryExce
 		}
 		return this.fieldbookMiddlewareService.countPlotsWithRecordedVariatesInDataset(this.userSelection.getWorkbook()
 				.getMeasurementDatesetId(), idParams);
+	}
 
+	@ResponseBody
+	@RequestMapping(value = "/checkForNonMaintenanceAndDerivativeMethods/{id}/{trialInstances}", method = RequestMethod.GET)
+	public Map<String, String> checkForNonMaintenanceAndDerivativeMethods(@PathVariable final String id, @PathVariable final Set<String> trialInstances) throws MiddlewareQueryException {
+		final Map<String, String> result = new HashMap<>();
+		final List<String> nonAdvancingMethods = this.studyDataManager.getNonMaintenanceAndDerivativeMethods(this.userSelection.getWorkbook()
```

Comment:
Make this function generic. Maybe instead of returning a list of string, return the actual Method? Rename this to getMethodsFromExperiments and parametarized the MethodType.